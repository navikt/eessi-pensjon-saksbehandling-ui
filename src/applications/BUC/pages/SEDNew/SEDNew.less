.a-buc-p-sednew {
  padding: 2rem 5rem 2rem 5rem !important;
}
