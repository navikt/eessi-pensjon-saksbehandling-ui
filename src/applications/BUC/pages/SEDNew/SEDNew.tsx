import SEDStart, { SEDStartProps } from 'applications/BUC/components/SEDStart/SEDStart'
import { BUCNewSelector } from 'applications/BUC/pages/BUCNew/BUCNew'
import { State } from 'declarations/reducers'
import { standardLogger, timeDiffLogger, timeLogger } from 'metrics/loggers'
import Panel from 'nav-frontend-paneler'
import { theme, themeHighContrast } from 'nav-styled-component-theme'
import React, { useEffect, useState } from 'react'
import { useSelector } from 'react-redux'
import styled, { ThemeProvider } from 'styled-components'

const SEDNewDiv = styled(Panel)`
  padding: 2rem 5rem 2rem 5rem !important;
  border: 1px solid ${({ theme }: any) => theme.navGra40};
  background-color: ${({ theme }: any) => theme['main-background-color']};
`

export interface SEDNewSelector {
  highContrast: boolean
}

const mapState = (state: State): SEDNewSelector => ({
  highContrast: state.ui.highContrast
})

const SEDNew: React.FC<SEDStartProps> = (props: SEDStartProps): JSX.Element => {
  const [loggedTime] = useState<Date>(new Date())
  const [totalTimeWithMouseOver, setTotalTimeWithMouseOver] = useState<number>(0)
  const [mouseEnterDate, setMouseEnterDate] = useState<Date | undefined>(undefined)
  const { highContrast } = useSelector<State, BUCNewSelector>(mapState)

  useEffect(() => {
    standardLogger('sed.new.entrance')
    return () => {
      timeLogger('sed.new.view', loggedTime)
      timeDiffLogger('sed.new.mouseover', totalTimeWithMouseOver)
    }
  })

  const onMouseEnter = () => setMouseEnterDate(new Date())

  const onMouseLeave = () => {
    if (mouseEnterDate) {
      setTotalTimeWithMouseOver(totalTimeWithMouseOver + (new Date().getTime() - mouseEnterDate?.getTime()))
    }
  }

  return (
    <ThemeProvider theme={highContrast ? themeHighContrast : theme}>
      <div
        onMouseLeave={onMouseLeave}
        onMouseEnter={onMouseEnter}
      >
        <SEDNewDiv>
          <SEDStart {...props} />
        </SEDNewDiv>
      </div>
    </ThemeProvider>
  )
}

export default SEDNew
