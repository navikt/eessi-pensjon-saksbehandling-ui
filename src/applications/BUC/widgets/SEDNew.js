import React from 'react'
import { Hovedknapp, Panel, Ingress } from 'components/ui/Nav'
import SEDStart from 'applications/BUC/components/SEDStart/SEDStart'

const SEDNew = (props) => {
  const { t, actions } = props

  const onBUCList = () => {
    actions.setMode('list')
  }

  return <React.Fragment>
    <div className='a-buc-buclist-buttons mb-2'>
      <div />
      <Hovedknapp onClick={onBUCList}>{t('buc:form-backToList')}</Hovedknapp>
    </div>
    <Panel>
      <Ingress>{t('buc:step-startSEDTitle')}</Ingress>
      <hr />
      <SEDStart {...props} />
    </Panel>
  </React.Fragment>
}

export default SEDNew
