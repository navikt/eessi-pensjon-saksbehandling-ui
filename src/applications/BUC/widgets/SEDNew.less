.a-buc-w-sednew {
  border: 1px solid #59514B;
  padding: 2rem 5rem 2rem 5rem !important;
}
