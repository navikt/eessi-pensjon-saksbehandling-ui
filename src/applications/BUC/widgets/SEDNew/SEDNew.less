.a-buc-w-sednew {
  padding: 2rem 5rem 2rem 5rem !important;
}
