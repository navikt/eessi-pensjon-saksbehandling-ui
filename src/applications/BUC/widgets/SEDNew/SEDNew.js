import React from 'react'
import { Panel } from 'components/ui/Nav'
import SEDStart from 'applications/BUC/components/SEDStart/SEDStart'
import './SEDNew.css'

const SEDNew = (props) => {
  return <Panel className='a-buc-sednew'>
    <SEDStart {...props} />
  </Panel>
}

export default SEDNew
