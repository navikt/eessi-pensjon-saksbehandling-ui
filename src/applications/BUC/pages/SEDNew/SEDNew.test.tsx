import React from 'react'
import SEDNew from './SEDNew'
import { mount, ReactWrapper } from 'enzyme'
jest.mock('applications/BUC/components/SEDStart/SEDStart', () => {
  return () => { return <div className='mock-sedstart' /> }
})

describe('applications/BUC/widgets/SEDNew/SEDNew', () => {
  let wrapper: ReactWrapper
  const initialMockProps = {
    bucs: {},
    setMode: jest.fn(),
    t: jest.fn(t => t)
  }
  it('Renders', () => {
    wrapper = mount(<SEDNew {...initialMockProps} />)
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Has proper HTML structure', () => {
    wrapper = mount(<SEDNew {...initialMockProps} />)
    expect(wrapper.exists('.a-buc-p-sednew')).toBeTruthy()
    expect(wrapper.exists('.mock-sedstart')).toBeTruthy()
  })
})
