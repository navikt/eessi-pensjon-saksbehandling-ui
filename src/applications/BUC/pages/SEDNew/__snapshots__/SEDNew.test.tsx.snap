// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/widgets/SEDNew/SEDNew Renders 1`] = `
<SEDNew
  bucs={Object {}}
  setMode={[MockFunction]}
>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Panel
      border={false}
      className="a-buc-p-sednew s-border"
    >
      <div
        className="panel a-buc-p-sednew s-border"
      >
        <Component
          bucs={Object {}}
          setMode={[MockFunction]}
        >
          <div
            className="mock-sedstart"
          />
        </Component>
      </div>
    </Panel>
  </div>
</SEDNew>
`;
